# Analysing Used iPhone Market

- The main aim of this project is to analyse the used iPhone market in Lithuania.

- The dataset used was scraped from a Lithuanian advertisement platform for primarily used items (Skelbiu) in December, 2025. The dataset describes used iPhone listings (views, likes, description) and phone attributes such as model, condition, price, etc.

### Libraries

In [ ]:
import pandas as pd

### Dataset

In [42]:
iphones = pd.read_csv('./iphones20251227.csv')

## Dataset overview

In [43]:
iphones

id                          name  maker              model  \
0     82852315.0       iPhone 15 pro max 256gb  Apple  iPhone 15 Pro Max   
1     82852217.0             iPhone 14 pro max  Apple  iPhone 14 Pro Max   
2     82852209.0                     iPhone 14  Apple          iPhone 14   
3     82852065.0           iPhone 14 pro 128gb  Apple      iPhone 14 Pro   
4     82852055.0               iPhone 16 128gb  Apple          iPhone 16   
...          ...                           ...    ...                ...   
3377  82507175.0  Parduodu, labai geros būklės  Apple          iPhone 12   
3378  82845939.0             Iphonas 11pro max  Apple  iPhone 11 Pro Max   
3379  82616093.0                  Geros bukles  Apple          iPhone 13   
3380  82822955.0                   Bukle 10/10  Apple  iPhone 13 Pro Max   
3381  82836327.0            Stovis labai geras  Apple      iPhone XS Max   

     condition  price         city  views  likes  \
0      Naudota  650.0      Vilnius    0.0    0.0   
1      Naudota  515.0     Šiauliai    2.0    0.0   
2      Naudota  300.0      Vilnius   11.0    1.0   
3      Naudota  470.0     Klaipėda    3.0    0.0   
4      Naudota  540.0     Šiauliai   15.0    3.0   
...        ...    ...          ...    ...    ...   
3377   Naudota  200.0  Marijampolė  227.0    2.0   
3378   Naudota  140.0      Vilnius   27.0    3.0   
3379   Naudota  200.0      Vilnius  123.0    5.0   
3380   Naudota  330.0       Kaunas   75.0    0.0   
3381   Naudota  120.0       Kaunas   77.0    2.0   

                                            description  \
0     iPhone 15 Pro Max 256GB, 89% baterija, puikios...   
1     Parduodamas „iPhone 14 Pro Max“. 256 GB, 84 % ...   
2     Iphone 14 128GB Blue. Baterija 90%. Pilnos kom...   
3     Parduodu iPhone 14 Pro 128 GB, telefonas tvark...   
4     Idealaus stovio. 128GB\nViskas originalu nieka...   
...                                                 ...   
3377                                      Zalias, 128gb   
3378                     Iphonas pro max 11 baterija 79   
3379                     jokiu ibrezimu veikia puikiai.   
3380                                 Baterija 86 128 gb   
3381                                 256 gb baterija 85   

                                                    url  stars   sold  \
0     https://www.skelbiu.lt/skelbimai/iphone-15-pro...    1.0  False   
1     https://www.skelbiu.lt/skelbimai/iphone-14-pro...    NaN  False   
2     https://www.skelbiu.lt/skelbimai/iphone-14-828...    1.0  False   
3     https://www.skelbiu.lt/skelbimai/iphone-14-pro...    NaN  False   
4     https://www.skelbiu.lt/skelbimai/iphone-16-128...    NaN  False   
...                                                 ...    ...    ...   
3377  https://www.skelbiu.lt/skelbimai/parduodu-laba...    NaN  False   
3378  https://www.skelbiu.lt/skelbimai/iphonas-11pro...    NaN  False   
3379  https://www.skelbiu.lt/skelbimai/geros-bukles-...    NaN  False   
3380  https://www.skelbiu.lt/skelbimai/bukle-10-10-8...    NaN  False   
3381  https://www.skelbiu.lt/skelbimai/stovis-labai-...    NaN  False   

     sale_time                last_update    registration_date  n_listings  \
0          NaN   Atnaujintas prieš 2 min.  2018-09-01 00:00:00        58.0   
1          NaN  Atnaujintas prieš 19 min.  2021-11-01 00:00:00        31.0   
2          NaN  Atnaujintas prieš 20 min.  2015-04-01 00:00:00         4.0   
3          NaN  Atnaujintas prieš 21 min.  2014-11-01 00:00:00       203.0   
4          NaN  Atnaujintas prieš 28 min.  2021-03-01 00:00:00        66.0   
...        ...                        ...                  ...         ...   
3377       NaN     Atnaujintas prieš 1 d.  2025-10-01 00:00:00         5.0   
3378       NaN     Atnaujintas prieš 1 d.  2021-11-01 00:00:00        16.0   
3379       NaN     Atnaujintas prieš 1 d.  2016-07-01 00:00:00        16.0   
3380       NaN  Atnaujintas prieš 10 val.  2021-02-01 00:00:00         8.0   
3381       NaN  Atnaujintas prie

There are no mistakes caused by reading the dataset. Everything looks normal.

In [44]:
iphones[iphones.duplicated()].head()

id                                    name  maker  \
24   82850887.0          iPhone 13 Pro ,Idealios Būklės  Apple   
60   82851417.0   iPhone 14 Pro Max. Battery Health 94%  Apple   
288  82822199.0                      iPhone 16pro 128gb  Apple   
312  82202426.0                               iPhone 14  Apple   
386  76432825.0  iPhone 16 Pro 256gb open box garantija  Apple   

                 model condition  price      city   views  likes  \
24       iPhone 13 Pro   Naudota  350.0  Klaipėda    31.0    3.0   
60   iPhone 14 Pro Max   Naudota  450.0   Vilnius    29.0    0.0   
288      iPhone 16 Pro   Naudota  480.0   Vilnius   267.0   10.0   
312          iPhone 14   Naudota  400.0  Klaipėda   136.0    2.0   
386      iPhone 16 Pro     Nauja  900.0   Vilnius  3000.0   38.0   

                                           description  \
24              Labai tvarkingas be ibrezimu.\nDežute.   
60   Parduodu puikiai išlaikytą iPhone 14 Pro Max 1...   
288  Parduodamas iphone 16pro 128gb , skilęs ekrana...   
312  Labai geros būklės Iphone 14 128GB\nBaterija 8...   
386  naujas, su garantija:\niPhone 16 Pro 256GB des...   

                                                   url  stars   sold  \
24   https://www.skelbiu.lt/skelbimai/iphone-13-pro...    1.0  False   
60   https://www.skelbiu.lt/skelbimai/iphone-14-pro...    1.0  False   
288  https://www.skelbiu.lt/skelbimai/iphone-16pro-...    NaN  False   
312  https://www.skelbiu.lt/skelbimai/iphone-14-822...    NaN  False   
386  https://www.skelbiu.lt/skelbimai/iphone-16-pro...    NaN  False   

    sale_time                last_update    registration_date  n_listings  \
24        NaN   Atnaujintas prieš 2 val.  2023-12-01 00:00:00         1.0   
60        NaN   Atnaujintas prieš 2 val.  2025-04-01 00:00:00        68.0   
288       NaN  Atnaujintas prieš 11 val.  2023-09-01 00:00:00       124.0   
312       NaN   Atnaujintas prieš 6 val.  2010-02-01 00:00:00      3328.0   
386       NaN   Atnaujintas prieš 7 val.  2014-10-01 00:00:00     10750.0   

    seller_verified  
24             True  
60            False  
288            True  
312            True  
386            True

There are quite a few rows that are duplicated. The website that was scraped sometimes shows ads that appeared is the last page. We will remove them.

In [45]:
iphones = iphones.drop_duplicates()

In [46]:
iphones.info()

<class 'pandas.core.frame.DataFrame'>
Index: 3342 entries, 0 to 3381
Data columns (total 18 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   id                 3342 non-null   float64
 1   name               3342 non-null   object 
 2   maker              3342 non-null   object 
 3   model              3342 non-null   object 
 4   condition          3340 non-null   object 
 5   price              3280 non-null   float64
 6   city               3225 non-null   object 
 7   views              3342 non-null   float64
 8   likes              3342 non-null   float64
 9   description        3162 non-null   object 
 10  url                3342 non-null   object 
 11  stars              94 non-null     float64
 12  sold               3342 non-null   bool   
 13  sale_time          117 non-null    object 
 14  last_update        3342 non-null   object 
 15  registration_date  3225 non-null   object 
 16  n_listings         3223 non-n

We are left with 3342 ads. Some columns have missings values: there are listings that did not provide a price, the vast majority of listings did not have any 'starts' (were not promoted by the seller) and so on.

### ID

In [47]:
iphones['id']

0       82852315.0
1       82852217.0
2       82852209.0
3       82852065.0
4       82852055.0
           ...    
3377    82507175.0
3378    82845939.0
3379    82616093.0
3380    82822955.0
3381    82836327.0
Name: id, Length: 3342, dtype: float64

Each listing is assigned a unique ID and it is visible on the website.

In [48]:
iphones[iphones['id'].duplicated()].shape[0]

20

Checking for futher duplicates based on ID, 20 rows come up. These also have to be removed.

In [49]:
iphones = iphones.drop_duplicates(subset='id', keep='last')

### Name

In [50]:
iphones['name']

0                                 iPhone 15 pro max 256gb
16                           iPhone 13 128gb 86%baterijos
17      Idelaus stovio iPhone 14 Pro Max Space Black 1...
18                  iPhone 14 Pro Max. Battery Health 94%
21                         iPhone 13 Pro ,Idealios Būklės
                              ...                        
3377                         Parduodu, labai geros būklės
3378                                    Iphonas 11pro max
3379                                         Geros bukles
3380                                          Bukle 10/10
3381                                   Stovis labai geras
Name: name, Length: 3322, dtype: object

Names do not necessarily describe what is being advertised. Some names are set as 'Conditions wery good' ('Stovis labai geras'), but others do have useful additional information that could be extracted such as storage capacity, color, battery health, etc.

### Model

In [53]:
iphones['model'].value_counts()

model
iPhone 13            276
iPhone 14 Pro        199
iPhone 11            194
iPhone 14 Pro Max    194
iPhone 13 Pro        181
iPhone 15 Pro Max    176
iPhone 12            153
iPhone 14            149
iPhone 13 Pro Max    142
iPhone 12 Pro        134
iPhone 15 Pro        130
iPhone 16 Pro Max    108
iPhone 12 Pro Max    105
iPhone 15            105
iPhone XR             73
iPhone 11 Pro         64
iPhone 11 Pro Max     61
iPhone 16 Pro         56
iPhone X              55
iPhone 12 mini        54
iPhone 8              50
iPhone 17 Pro Max     48
iPhone 7              45
iPhone 13 mini        45
iPhone SE (2020)      45
iPhone 16             44
iPhone 14 Plus        43
iPhone 17 Pro         39
iPhone 6S             37
iPhone XS             35
iPhone 8 Plus         32
iPhone XS Max         31
iPhone 15 Plus        29
iPhone 7 Plus         26
iPhone SE             20
iPhone 16e            19
iPhone 5S             17
iPhone 17             17
iPhone 6              16
iPhone 16 Plus     

In [54]:
iphones['model'].unique()

array(['iPhone 15 Pro Max', 'iPhone 13', 'iPhone 14 Pro Max',
       'iPhone 13 Pro', 'iPhone 14 Pro', 'iPhone 15 Pro',
       'iPhone 13 Pro Max', 'iPhone 17 Pro Max', 'iPhone 11',
       'iPhone 16 Pro Max', 'iPhone 12 Pro Max', 'iPhone 14 Plus',
       'iPhone 15', 'iPhone 15 Plus', 'iPhone SE', 'iPhone 17',
       'iPhone 11 Pro', 'iPhone 14', 'iPhone 16 Pro', 'iPhone 12 Pro',
       'iPhone 13 mini', 'iPhone 16', 'iPhone X', 'iPhone 12 mini',
       'iPhone 5S', 'iPhone 6S', 'iPhone 12', 'iPhone XS', 'iPhone 16e',
       'iPhone 8 Plus', 'iPhone XR', 'iPhone 7', 'iPhone 5', 'iPhone 4S',
       'iPhone 6', 'iPhone 8', 'iPhone 3GS', 'iPhone SE (2020)',
       'iPhone 17 Pro', 'iPhone XS Max', 'iPhone 7 Plus', 'iPhone Air',
       'iPhone 3G', 'iPhone 16 Plus', 'iPhone 6S Plus',
       'iPhone 11 Pro Max', 'iPhone SE (2022)', 'iPhone 6 Plus',
       'iPhone 4', 'iPhone 5C'], dtype=object)

Practically all models are available, going back as far as the iPhone 4. The most common listing is for the iPhone 13, closely followed by two iPhone 14 variants and iPhone 11. Grouping iPhone variants into bins could paint a clearer picture.

In [55]:
model_mapping = {
    "iPhone 15 Pro Max": "iPhone 15",
    "iPhone 13": "iPhone 13",
    "iPhone 14 Pro Max": "iPhone 14",
    "iPhone 13 Pro": "iPhone 13",
    "iPhone 14 Pro": "iPhone 14",
    "iPhone 15 Pro": "iPhone 15",
    "iPhone 13 Pro Max": "iPhone 13",
    "iPhone 17 Pro Max": "iPhone 17",
    "iPhone 11": "iPhone 11",
    "iPhone 16 Pro Max": "iPhone 16",
    "iPhone 12 Pro Max": "iPhone 12",
    "iPhone 14 Plus": "iPhone 14",
    "iPhone 15": "iPhone 15",
    "iPhone 15 Plus": "iPhone 15",
    "iPhone SE": "iPhone SE",
    "iPhone 17": "iPhone 17",
    "iPhone 11 Pro": "iPhone 11",
    "iPhone 14": "iPhone 14",
    "iPhone 16 Pro": "iPhone 16",
    "iPhone 12 Pro": "iPhone 12",
    "iPhone 13 mini": "iPhone 13",
    "iPhone 16": "iPhone 16",
    "iPhone X": "iPhone X",
    "iPhone 12 mini": "iPhone 12",
    "iPhone 5S": "iPhone 5",
    "iPhone 6S": "iPhone 6",
    "iPhone 12": "iPhone 12",
    "iPhone XS": "iPhone X",
    "iPhone 16e": "iPhone 16",
    "iPhone 8 Plus": "iPhone 8",
    "iPhone XR": "iPhone X",
    "iPhone 7": "iPhone 7",
    "iPhone 5": "iPhone 5",
    "iPhone 4S": "iPhone 4",
    "iPhone 6": "iPhone 6",
    "iPhone 8": "iPhone 8",
    "iPhone 3GS": "iPhone 3",
    "iPhone SE (2020)": "iPhone SE",
    "iPhone 17 Pro": "iPhone 17",
    "iPhone XS Max": "iPhone X",
    "iPhone 7 Plus": "iPhone 7",
    "iPhone Air": "iPhone Air",
    "iPhone 3G": "iPhone 3",
    "iPhone 16 Plus": "iPhone 16",
    "iPhone 6S Plus": "iPhone 6",
    "iPhone 11 Pro Max": "iPhone 11",
    "iPhone SE (2022)": "iPhone SE",
    "iPhone 6 Plus": "iPhone 6",
    "iPhone 4": "iPhone 4",
    "iPhone 5C": "iPhone 5"
}

In [56]:
iphones = iphones.assign(base_model = iphones['model'].map(model_mapping))